diff -urN dsniff-2.4.org/arpspoof.c dsniff-2.4/arpspoof.c
--- dsniff-2.4.org/arpspoof.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/arpspoof.c	2003-11-08 22:45:50.000000000 +0100
@@ -19,7 +19,7 @@
 #include <string.h>
 #include <signal.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <pcap.h>
 
 #include "arp.h"
diff -urN dsniff-2.4.org/configure.in dsniff-2.4/configure.in
--- dsniff-2.4.org/configure.in	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/configure.in	2003-11-08 22:46:09.000000000 +0100
@@ -204,43 +204,43 @@
 AC_SUBST(PCAPINC)
 AC_SUBST(PCAPLIB)
 
-dnl Checks for libnet
-AC_MSG_CHECKING(for libnet)
-AC_ARG_WITH(libnet,
-[  --with-libnet=DIR       use libnet in DIR],
+dnl Checks for libnet1
+AC_MSG_CHECKING(for libnet1)
+AC_ARG_WITH(libnet1,
+[  --with-libnet1=DIR       use libnet1 in DIR],
 [ case "$withval" in
   yes|no)
      AC_MSG_RESULT(no)
      ;;
   *)
      AC_MSG_RESULT($withval)
-     if test -f $withval/include/libnet.h -a -f $withval/lib/libnet.a; then
+     if test -f $withval/include/libnet1.h -a -f $withval/lib/libnet1.a; then
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
-	if test -f $withval/bin/libnet-config; then
-	   CFLAGS="$CFLAGS `$withval/bin/libnet-config --defines`"
-	elif test -f $withval/libnet-config; then
-	   CFLAGS="$CFLAGS `$withval/libnet-config --defines`"
+	if test -f $withval/bin/libnet1-config; then
+	   CFLAGS="$CFLAGS `$withval/bin/libnet1-config --defines`"
+	elif test -f $withval/libnet1-config; then
+	   CFLAGS="$CFLAGS `$withval/libnet1-config --defines`"
 	else
-	   CFLAGS="$CFLAGS `libnet-config --defines`"
+	   CFLAGS="$CFLAGS `libnet1-config --defines`"
 	fi
 	LNETINC="-I$withval/include"
-	LNETLIB="-L$withval/lib -lnet"
+	LNETLIB="-L$withval/lib -lnet1"
      else
-        AC_ERROR(libnet.h or libnet.a not found in $withval)
+        AC_ERROR(libnet1.h or libnet1.a not found in $withval)
      fi
      ;;
   esac ],
-[ if test -f ${prefix}/include/libnet.h; then
-     CFLAGS="$CFLAGS `${prefix}/bin/libnet-config --defines`"
+[ if test -f ${prefix}/include/libnet1.h; then
+     CFLAGS="$CFLAGS `${prefix}/bin/libnet1-config --defines`"
      LNETINC="-I${prefix}/include"
-     LNETLIB="-L${prefix}/lib -lnet"
-  elif test -f /usr/include/libnet.h; then
-     CFLAGS="$CFLAGS `libnet-config --defines`"
-     LNETLIB="-lnet"
+     LNETLIB="-L${prefix}/lib -lnet1"
+  elif test -f /usr/include/libnet1.h; then
+     CFLAGS="$CFLAGS `libnet1-config --defines`"
+     LNETLIB="-lnet1"
   else
      AC_MSG_RESULT(no)
-     AC_ERROR(libnet not found)
+     AC_ERROR(libnet1 not found)
   fi
   AC_MSG_RESULT(yes) ]
 )
diff -urN dsniff-2.4.org/decode_portmap.c dsniff-2.4/decode_portmap.c
--- dsniff-2.4.org/decode_portmap.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/decode_portmap.c	2003-11-08 22:45:50.000000000 +0100
@@ -16,7 +16,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 
 #include "rpc.h"
diff -urN dsniff-2.4.org/dnsspoof.c dsniff-2.4/dnsspoof.c
--- dsniff-2.4.org/dnsspoof.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/dnsspoof.c	2003-11-08 22:45:50.000000000 +0100
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <resolv.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <pcap.h>
 
 #include "pcaputil.h"
diff -urN dsniff-2.4.org/dsniff.c dsniff-2.4/dsniff.c
--- dsniff-2.4.org/dsniff.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/dsniff.c	2003-11-08 22:45:51.000000000 +0100
@@ -19,7 +19,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 #include <pcap.h>
 
diff -urN dsniff-2.4.org/filesnarf.c dsniff-2.4/filesnarf.c
--- dsniff-2.4.org/filesnarf.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/filesnarf.c	2003-11-08 22:45:51.000000000 +0100
@@ -19,7 +19,7 @@
 #include <fcntl.h>
 #include <regex.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 #include <pcap.h>
 
diff -urN dsniff-2.4.org/macof.c dsniff-2.4/macof.c
--- dsniff-2.4.org/macof.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/macof.c	2003-11-08 22:45:51.000000000 +0100
@@ -20,7 +20,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <pcap.h>
 
 #include "version.h"
diff -urN dsniff-2.4.org/mailsnarf.c dsniff-2.4/mailsnarf.c
--- dsniff-2.4.org/mailsnarf.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/mailsnarf.c	2003-11-08 22:45:51.000000000 +0100
@@ -19,7 +19,7 @@
 #include <time.h>
 #include <regex.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 #include <pcap.h>
 
diff -urN dsniff-2.4.org/msgsnarf.c dsniff-2.4/msgsnarf.c
--- dsniff-2.4.org/msgsnarf.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/msgsnarf.c	2003-11-08 22:45:51.000000000 +0100
@@ -19,7 +19,7 @@
 #include <string.h>
 #include <regex.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 #include <pcap.h>
 #include <pcaputil.h>
diff -urN dsniff-2.4.org/record.c dsniff-2.4/record.c
--- dsniff-2.4.org/record.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/record.c	2003-11-08 22:45:51.000000000 +0100
@@ -21,7 +21,7 @@
 #elif HAVE_DB_H
 #include <db.h>
 #endif
-#include <libnet.h>
+#include <libnet1.h>
 
 #include "options.h"
 #include "record.h"
diff -urN dsniff-2.4.org/sshmitm.c dsniff-2.4/sshmitm.c
--- dsniff-2.4.org/sshmitm.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/sshmitm.c	2003-11-08 22:45:51.000000000 +0100
@@ -16,7 +16,7 @@
 #include <sys/wait.h>
 #include <arpa/nameser.h>
 #include <openssl/ssl.h>
-#include <libnet.h>
+#include <libnet1.h>
 
 #include <err.h>
 #include <errno.h>
diff -urN dsniff-2.4.org/tcpkill.c dsniff-2.4/tcpkill.c
--- dsniff-2.4.org/tcpkill.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/tcpkill.c	2003-11-08 22:45:51.000000000 +0100
@@ -17,7 +17,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <pcap.h>
 
 #include "pcaputil.h"
diff -urN dsniff-2.4.org/tcpnice.c dsniff-2.4/tcpnice.c
--- dsniff-2.4.org/tcpnice.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/tcpnice.c	2003-11-08 22:45:51.000000000 +0100
@@ -17,7 +17,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <pcap.h>
 
 #include "pcaputil.h"
diff -urN dsniff-2.4.org/tcp_raw.c dsniff-2.4/tcp_raw.c
--- dsniff-2.4.org/tcp_raw.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/tcp_raw.c	2003-11-08 22:45:51.000000000 +0100
@@ -14,7 +14,7 @@
 
 #include <time.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include "options.h"
 #include "tcp_raw.h"
 
diff -urN dsniff-2.4.org/trigger.c dsniff-2.4/trigger.c
--- dsniff-2.4.org/trigger.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/trigger.c	2003-11-08 22:45:51.000000000 +0100
@@ -14,7 +14,7 @@
 
 #include <stdlib.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 
 #include "decode.h"
diff -urN dsniff-2.4.org/urlsnarf.c dsniff-2.4/urlsnarf.c
--- dsniff-2.4.org/urlsnarf.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/urlsnarf.c	2003-11-08 22:45:51.000000000 +0100
@@ -22,7 +22,7 @@
 #include <regex.h>
 #include <time.h>
 #include <err.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 #include <pcap.h>
 
diff -urN dsniff-2.4.org/webmitm.c dsniff-2.4/webmitm.c
--- dsniff-2.4.org/webmitm.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/webmitm.c	2003-11-08 22:45:51.000000000 +0100
@@ -20,7 +20,7 @@
 
 #include <err.h>
 #include <errno.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -urN dsniff-2.4.org/webspy.c dsniff-2.4/webspy.c
--- dsniff-2.4.org/webspy.c	2003-11-08 22:37:23.000000000 +0100
+++ dsniff-2.4/webspy.c	2003-11-08 22:45:51.000000000 +0100
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <err.h>
 #include <X11/Xlib.h>
-#include <libnet.h>
+#include <libnet1.h>
 #include <nids.h>
 
 #include "base64.h"
